tests/management/principal/test_it_service.py

```diff
@@ -0,0 +1,334 @@
+        principal_1 = Principal.objects.create(username="user-1", tenant=self.tenant)
+        principal_2 = Principal.objects.create(username="user-2", tenant=self.tenant)
+        principal_3 = Principal.objects.create(username="user-3", tenant=self.tenant)
+
+        client_uuid_1 = uuid.uuid4()
+        client_uuid_2 = uuid.uuid4()
+        client_uuid_3 = uuid.uuid4()
+        client_uuid_4 = uuid.uuid4()
+        client_uuid_5 = uuid.uuid4()
+        sa_1 = Principal.objects.create(
+            username=f"service-account-{client_uuid_1}",
+            service_account_id=client_uuid_1,
+            type="service-account",
+            tenant=self.tenant,
+        )
+        sa_2 = Principal.objects.create(
+            username=f"service-account-{client_uuid_2}",
+            service_account_id=client_uuid_2,
+            type="service-account",
+            tenant=self.tenant,
+        )
+        sa_3 = Principal.objects.create(
+            username=f"service-account-{client_uuid_3}",
+            service_account_id=client_uuid_3,
+            type="service-account",
+            tenant=self.tenant,
+        )
+        sa_4 = Principal.objects.create(
+            username=f"service-account-{client_uuid_4}",
+            service_account_id=client_uuid_4,
+            type="service-account",
+            tenant=self.tenant,
+        )
+        sa_5 = Principal.objects.create(
+            username=f"service-account-{client_uuid_5}",
+            service_account_id=client_uuid_5,
+            type="service-account",
+            tenant=self.tenant,
+        )
+
+        # Create a set with the service accounts that will go in the group. It will make it easier to make assertions
+        # below.
+        group_service_accounts_set = {
+            str(sa_1.service_account_id),
+            str(sa_2.service_account_id),
+            str(sa_3.service_account_id),
+            str(sa_4.service_account_id),
+            str(sa_5.service_account_id),
+        }
+
+        # For this test, we don't associate any service accounts to it.
+        group = Group(name="it-service-group", platform_default=False, system=False, tenant=self.tenant)
+        group.save()
+        # Add the principal accounts to make sure that we are only working with service accounts. If we weren't, these
+        # principals below should give us unexpected results in our assertions.
+        group.principals.add(principal_1)
+        group.principals.add(principal_2)
+        group.principals.add(principal_3)
+        # Add the service accounts to the group.
+        group.principals.add(sa_1)
+        group.principals.add(sa_2)
+        group.principals.add(sa_3)
+        group.principals.add(sa_4)
+        group.principals.add(sa_5)
+        group.save()
+
+        # Simulate that a few client IDs were specified in the request.
+        request_client_ids = set[str]()
```

Comment:
the `request_client_ids` is initialised as `set` of strings and then the `UUID` instances are added

Reply:
Refactored!